### DISCUSSION TITLE
Add support for cards

### DISCUSSION THREAD

**Discussion by ilg-ul** [jothepro/doxygen-awesome-css#117] [Ideas]
My evaluation of Doxygen + the Awesome theme went well, I ended up moving all the content from various READMEs do web pages, and the project currently has a minimal README referring to the documentation web:

- <https://github.com/micro-os-plus/utils-lists-xpack/tree/xpack-develop>

To make the home page even nicer, I tried to add what are usually called _cards_, rectangles with a short text, linking to the actual content.

The result currently looks like this:

![Screenshot 2023-05-19 at 21 01 18](https://github.com/jothepro/doxygen-awesome-css/assets/362732/5455e3b6-7fa1-4c90-947d-d6b1358e81d9)

which, in my opinion, looks nicer than the usual plain sections.

If you find this a useful feature, perhaps we can add the CSSs to the theme, and a small example in the documentation on how to use it.

Unfortunately my knowledge of CSS is not good enough to suggest a proper PR, my code is barely functional, and it might even be incorrect.

Any thought about this proposal?

**Comment by jothepro** [comment]
In general I am a bit hesitant regarding adding features that are not directly related to styling Doxygen and that are easily added with some custom CSS/HTML. Take the "Next Page" buttons from the project documentation as an example:
<img width="130" alt="Next Page Button" src="https://github.com/jothepro/doxygen-awesome-css/assets/21294002/bc16a062-feb0-404d-89aa-bf338d9063db">

They are not officially part of the custom CSS, nevertheless I have seen people adapting the concept on their documentation pages.

For now I think you should share your concept in [Custom User Themes](https://github.com/jothepro/doxygen-awesome-css/discussions/13) to allow others to give feedback and adapt the concept.
That being said, I really like the idea and I will keep it in mind. 👍 
Maybe there is a smart way to make them Markdown compatible without requiring that much custom HTML. In that case I'd be intrigued to add them... 🤔

**Reply by ilg-ul** [reply]
> For now I think you should share your concept in https://github.com/jothepro/doxygen-awesome-css/discussions/13 to allow others to give feedback and adapt the concept.

Great suggestion, I added this: https://github.com/jothepro/doxygen-awesome-css/discussions/13#discussioncomment-5980507.

> I really like the idea and I will keep it in mind.

Thank you for your appreciation :-)

> Maybe there is a smart way to make them Markdown compatible without requiring that much custom HTML. In that case I'd be intrigued to add them...

That would be an interesting idea, but what might it be? 

I was also thinking of adding a small image before the card title. How would you define a group with an image, a title and a paragraph in Doxygen?

**Reply by jothepro** [reply]
The only way that I can think of rn is doing some Javascript trickery like I did with the ["Tabs" extension](https://jothepro.github.io/doxygen-awesome-css/md_docs_extensions.html#extension-tabs): A markdown list is annotated with some class attributes, that can then be picked up by Javascript and transformed the the required HTML nodes. But on the other hand I'm not sure if it's a good idea to build more of these fancy polyfills because they are kind of fragile. I guess it would still be smarter to just add some custom HTML like you did, or to contribute an actual Markdown extension to Doxygen, so the required HTML can be generated when the docs are rendered.

**Comment by ilg-ul** [comment]
I added some small SVG images to the cards, I think this looks even nicer:

![Screenshot 2023-05-26 at 22 40 34](https://github.com/jothepro/doxygen-awesome-css/assets/362732/b9cf1728-6c6f-4660-b291-74fcee4f700a)
